src/apply_rules.rs

```diff
@@ -858,6 +883,8 @@ pub fn apply_rules(
             &mut collections.networks,
             networks_consolidation,
         )?;
+
+        update_ticket_use_perimeters(&mut collections, networks_consolidation.clone());
```

Comment:
hum I'm not sure you want to clone the networks here, you can just take a reference or an iterator.

If you take a reference, you can, in the signature of the function, take a slice instead of a `&Vec`:
```rust
fn update_ticket_use_perimeters(
    collections: &mut Collections,
    networks_consolidation: &[NetworkConsolidation],
```